src/savegame_json.cpp

```diff
@@ -3435,9 +3435,20 @@ void submap::store( JsonOut &jsout ) const
         jsout.end_array();
     }
     jsout.end_array();
-    // Output the computer
-    if( comp != nullptr ) {
-        jsout.member( "computers", comp->save_data() );
+
+    if( legacy_computer ) {
+        // it's possible that no access to computers has been made and legacy_computer
+        // is not cleared
+        jsout.member( "computers", legacy_computer->save_data() );
+    } else if( !computers.empty() ) {
+        jsout.member( "computers" );
+        jsout.start_array();
+        for( auto &elem : computers ) {
+            jsout.write( elem.first.x );
+            jsout.write( elem.first.y );
```

Comment:
Why write x and y manually? You can just serialize the `point` object directly.

Reply:
I didn't know that, it was done this way in the other functions, will look into it.